website/src/views/DashboardRouter/DashboardRouter.jsx

```diff
@@ -70,24 +73,87 @@ class DashboardRouter extends PureComponent {
     });
   };
 
+  viewEvent = (eventData) => {
+    if (navigator.onLine) {
+      message.info('Getting event...');
+      const { userData } = this.state;
+      const { eventUid, classUid, university } = eventData;
+      const { uid } = firebase.auth().currentUser;
+      const reqData = { eventUid, classUid, uid };
+      fetch(
+        'https://us-central1-rails-students.cloudfunctions.net/geteventdetails',
+        {
+          method: 'POST',
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(reqData),
+        },
+      )
+        .then(res => res.json())
+        .then((result) => {
+          message.info(result.message);
+          console.log(result);
+          if (result.message === 'Found event') {
+            result.eventData.university = university;
+            result.eventData.classUid = classUid;
+            result.eventData.eventUid = eventUid;
+            if (userData.type === 'teacher' && (result.eventData.instructorUid === uid)) {
+              this.setState({
+                visible: true,
+                eventData: result.eventData,
+              });
+            } else {
+              this.setState({
+                eventVisible: true,
+                eventData: result.eventData,
+              });
+            }
+          }
+        })
+        .catch((err) => {
+          console.log(err);
+          message.error(err.message);
+        });
+    } else {
+      message.error('Can\'t get event when offline');
```

Comment:
Arent events cached and you would show the cached event details

Reply:
unsafe because data for subsequent api calls wouldnt be fetched so its better to just deny access at this point. maybe if we had a more complex service worker implemented

Reply:
Based on how its currently implemented  in master it only doesn't fetch if the browser is online otherwise it will always fetch and cache.

```
if (navigator.onLine) {
      if (firebase.auth().currentUser !== undefined) {
        firebase.auth().currentUser.getIdToken(true)
          .then((idToken) => {
            const { uid } = firebase.auth().currentUser;
            const reqData = { uid, idToken };
            fetch(
              'https://us-central1-rails-students.cloudfunctions.net/getprofile',
              {
                method: 'POST',
                headers: {
                  Accept: 'application/json',
                  'Content-Type': 'application/json',
                },
                body: JSON.stringify(reqData),
              },
            )
              .then(res => res.json())
              .then((data) => {
                localStorage.setItem('DashboardRouter-teacher', (data.userData.type === 'teacher').toString());
                localStorage.setItem('DashboardRouter-userData', JSON.stringify(data.userData));
                this.setState({
                  teacher: data.userData.type === 'teacher',
                  userData: data.userData ? data.userData : {},
                });
              })
              .catch((err) => {
                console.log(err);
              });
          });
      }
    } else {
      this.setState({
        teacher: localStorage.getItem('DashboardRouter-teacher') === 'true',
        userData: JSON.parse(localStorage.getItem('DashboardRouter-userData')),
      });
    }
```

Reply:
getevent api returns more data than the getprofile api. the service worker is only caching userData. eventDat is separate